### DISCUSSION TITLE
Configuring options for resolving pie chart/donut labels' cutting and overlapping [Grafana 8.3.4]

### DISCUSSION THREAD

**Discussion by rohit-innodeed** [grafana/grafana#50563] [Dashboards, Panels & Visualizations]
[Grafana 8.3.4]

There are some issues with pie chart/donut labels which can be resolved by adding below features:

Feature 1: 
If the labels are long then they usually get cut as shown in the below image:

![image](https://user-images.githubusercontent.com/101693459/172995781-207a06bd-7f25-456c-9924-e164e0ccf9b9.png)

However this issue can be resolved just by overriding the 'overflow' property in grafana's root css using HTML plugin. Following are the steps:

1. Adding code in the root CSS block:

![image](https://user-images.githubusercontent.com/101693459/172996431-a441674e-5afb-491f-80b9-883c5b0ebdaf.png)

2. Checking whether the property is getting overrided or not using browser's inspect feature:

![image](https://user-images.githubusercontent.com/101693459/172997008-b42a4380-ad61-4bad-8909-0f267a96b882.png)

After following above steps, the labels don't get cut as shown in the below image:

![image](https://user-images.githubusercontent.com/101693459/172997725-7bb5a127-87f3-4358-affc-f42869325511.png)

So there can be an option for this for making 'overflow' property hidden or visible. 

Feature 2:
If there are more labels like Percent or Value along with Name then the labels start overlapping as shown in figure:

![image](https://user-images.githubusercontent.com/101693459/172999784-3b5d9f94-de0f-4c89-9ab0-fdc5778f9c35.png)

It will be good if there will be a feature for aligning the labels vertically or horizontally as shown in the below image:

![image](https://user-images.githubusercontent.com/101693459/173000350-3cd49fbc-8191-45a4-b263-a32a70530d8e.png)

Which will result in horizontal labels as shown in the  below image:

![image](https://user-images.githubusercontent.com/101693459/173000920-0435f4b5-1fde-4fe3-8c54-b26dcf9cc255.png)

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
